### PR TITLE
Prodigy clickable layout

### PR DESCRIPTION
**Pull request by JoakimLarsson**
The chess computer can now be used for playing a game of chess using the keypad and the BCD display

### CODE REVIEW

**Comment by happppp** [comment]
A browser UI via http. Sure sounds like fun :) but IMO doesn't have a place
in a MAME driver.

Unrelated to this PR but I'm curious: While I was reading your prodigy.cpp
driver a bit, I noticed you've made a netlist for the 7seg display. Wasn't
it possible to emulate accurately at higher level?

**Comment by JoakimLarsson** [comment]
The support for browser UI is purely a translation of the .LAY file to SVG using a generic XSLT transformation on the fly that I am working on, heck the XSLT reference at the top of the .LAY file. It would allow for extra entities in the .LAY file that only applies to SVG, like rounded rect corners and animations of chess pieces. I am not sure it will make it through either but then I can do the translation statically and just go from there.

The reason I did the netlist was because my 74164 PR was rejected because of a clock domain issue revealed by JD in the work with Hazeltime. I am not doing HLE unless I lack information or CPU cycles :)

**Comment by rb6502** [comment]
We have several drivers already that offer an HTTP UI in esq5505; Micko explicitly OKed them.  I agree that it's not something we necessarily want widespread unless we completely ditch the artwork system, but if this is a better UX than the current artwork-based chessboard it might be worth giving wider support to.

**Comment by JoakimLarsson** [comment]
Fingers crossed, but that will be a later PR, the http part is disabled in this PR, I just kept it in there to reduce merging efforts later when I get back to this driver and also to get some comments :)

File: src/mame/drivers/prodigy.cpp [hunk 1]
```diff
@@ -99,6 +120,16 @@ class prodigy_state : public driver_device
 		, m_cb1(*this, "bcd:cb1")
 		, m_cb2(*this, "bcd:cb2")
 		, m_digit(0.0)
+		, m_io_line0(*this, "LINE0")
+		, m_io_line1(*this, "LINE1")
+		, m_io_line2(*this, "LINE2")
+		, m_io_line3(*this, "LINE3")
+		, m_io_line4(*this, "LINE4")
```
Comment (cuavas) [inline src/mame/drivers/prodigy.cpp]: Is there a reason not to use `required_ioport_array` for this?

Comment (JoakimLarsson) [reply src/mame/drivers/prodigy.cpp]: no idea, will check required_ioport_array out, the refactoring I am planning to do will make it a 2x10 matrix as the keypad is wired that way which I didn't know when starting out.

**Comment by JoakimLarsson** [comment]
I'll keep the http implementation in the driver but I write so it can be moved later if approach is liked. I already know that I need a helper method in render.cpp that returns the decompressed layout and also setup a global /layout handler in the machine init or something.

The idea is to enhance the LAY file with information that helps generation of SVG and WebGL for a more accurate mechanical modelling and also add a websocket interface/protocol for interactions with the driver. But that is down the line and at a global level it would be nice to select driver too through this protocol but I need to think more about that part and there are probably others that has already.

**Pull request merged**